[STT-1223] Add commands to remap STT metadata fields

eos87 opened the pull request:
### Purpose
Introduce management commands to remap legacy STT metadata fields according to details requested in [STT-1223]. This work prepares the data for consistent filtering, list views, and templates as described in the ticket.

### What has changed
- Added a new `stt.commands` package and plugged it into `manage.py`

One new commands added:
~- `remap_stt_language`~
~This goes through items and agenda entries, and updates their language field. By default everything is set to Finnish, except for special cases like headlines ending in ***TRANSLATED*** or containing “News in brief” or “NEWS BULLETIN”, which stay in English. The command can be run in dry-run mode, limited to a certain number of items, and processes items in batches of 100.~

- `remap_stt_metadata`
This takes care of the other STT metadata clean-up. It clears out old service values (including “Australian General News”), maps `sttdepartment` into the `service` field (falling back to `Kotimaa` + `Pika+` when missing), and replaces old `sttsubj` codes with the new Media Topics vocabulary defined in [STT-1210]. ~For planning items, it also converts `stturgency` into `priority`.~ It can run in dry-run mode, print verbose output, and work in batches so it’s safe to run on large datasets.

Resolves: [STT-1223]

[STT-1223]: https://sofab.atlassian.net/browse/STT-1223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[STT-1210]: https://sofab.atlassian.net/browse/STT-1210?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Add STT language remapping management command

Introduces a new management command 'remap_stt_language' for remapping the language field in 'items' and 'agenda' resources based on headline content. Updates manage.py to register the new command module.

STT-1223

eos87 committed:
Refactor item iteration logic into utils module

Moved the item iteration and batching logic from remap_stt_metadata.py to a new iterate_items function in utils.py. This improves code reuse and readability in remap_stt_metadata.py by delegating resource iteration to a utility function.

STT-1223

eos87 committed:
Rename remap_stt_metadata to remap_stt_language

Renamed the module and updated the import in __init__.py to reflect the new name.

STT-1223

eos87 committed:
Add `remap_stt_metadata` command for category remapping

Introduces a new management command to remap 'sttdepartment' to 'anpa_category' for Wire and Agenda items.

STT-1223

eos87 committed:
Add STT media topics data and update remap script

Introduces a new `STT_Media_Topics.json` file containing a list of media topics according to ticket STT-1210.
Updates `remap_stt_metadata.py` to map the `sttsubj` with new media topics.

STT-1223

eos87 committed:
Refactor STT metadata remapping and add priority update

Replaces DefaultCategory dataclass with constants, adds update_service and update_priority functions, and refactors remap_stt_metadata to update service and priority fields. Improves batch size handling and enhances documentation for the remapping process.

STT-1223

eos87 committed:
Refactor `remap_stt_language` to remove utils dependency

Replaces usage of `iterate_items` from `utils.py` with direct batch processing using `service.get_all_batch` in `remap_stt_language.py`. Removes the now-unneeded `utils.py` file and adds batch size and sleep logic directly to `remap_stt_language` for improved clarity and maintainability.

STT-1223

eos87 committed:
Add tests for `remap_stt_language` and `remap_stt_metadata`

Introduces unit tests for remap_stt_language.py and remap_stt_metadata.py, including a FakeService mock for resource operations. Also updates remap_stt_language to remove a default argument and remap_stt_metadata to use 'mediatopic' as the subject scheme.

STT-1223

eos87 committed:
Fix black

petrjasek commented on server/stt/commands/remap_stt_metadata.py:
```diff
@@ -0,0 +1,157 @@
+with open(topics_cv_path, "r", encoding="utf-8") as f:
+    topics_cv = json.load(f)
+topics_map = {
+    item.get("iptc_subject"): item
```
I've noticed when working with that CV that the `iptc_subject` values are missing the leading zeros, 
which was making my values not match. what I did as a workaround was converting it to integer for the lookups
so eg there will be subject with code `06000000` and the iptc_subject value will be `6000000`

eos87 replied:
Good idea. Actually, that's why I added the second map with names as a fallback, but with your suggestion I probably won't need it.

eos87 replied:
Done. Still keep the fallback by names because there are a good amount that match exactly by name but do not have value on `iptc_subject` field

eos87 committed:
Improve subject mapping and log missing topics

Changed the subject mapping to use integer keys for 'iptc_subject' and 'code' fields. Added tracking and verbose logging of topics not found during remapping to aid debugging and data completeness.

eos87 committed:
Fix black

eos87 committed:
Adjust tests

STT-1223

eos87 commented on server/stt/commands/remap_stt_metadata.py:
```diff
@@ -0,0 +1,157 @@
+    for entry in subject:
+        if entry.get("scheme") == "sttdepartment":
+            updates["anpa_category"] = [
+                {"qcode": entry.get("code"), "name": entry.get("name")}
```
this is just `code`

eos87 commented on server/stt/commands/remap_stt_metadata.py:
```diff
@@ -0,0 +1,157 @@
+                updates["priority"] = int(entry.get("code")[-1])
+                return
+            except Exception:
+                continue
```
no longer needed

eos87 committed:
Remove priority update, adjust batch size to 500

Removed the update_priority function and its usage for agenda planning items. Increased the default limit to 1000 and batch size to 500 for improved processing efficiency. Also standardized the `anpa_category` field to use `code` instead of `qcode`.

STT-1223

eos87 committed:
Increase default limit and batch size in remap_stt_language

Raised the default --limit from 500 to 1000 and BATCH_SIZE from 100 to 500 in the remap_stt_language command to allow processing larger data sets per run.

eos87 committed:
Consolidate STT language remapping into metadata command

Removed the separate remap_stt_language command and its tests, moving language remapping logic into remap_stt_metadata. Updated remap_stt_metadata and its tests to handle language field updates for items and agenda resources, ensuring consistent processing and reducing code duplication.

STT-1223

eos87 committed:
Refactor service field mapping in STT metadata remap

- Replaces mapping of `sttdepartment` to `anpa_category` with mapping to `service` in remap_stt_metadata.py.
- Updates related tests to assert correct behavior for the `service` field and removes references to `anpa_category`.

STT-1223

eos87 committed:
Remove debug print statements from remap_stt_metadata.py

STT-1223

eos87 commented:
Hi @petrjasek, 
I've made a few changes after what we talked with Gideon: 
- Mapped `sttdepartment` to `service` field
- Removed the language command as it seemed redundant and integrated its changes into the other command
- Increased the batch size from 100 to 500 
- Removed the update of priority field
Please take another look when you have some time. Thanks!

petrjasek commented on server/stt/commands/remap_stt_metadata.py:
```diff
@@ -0,0 +1,157 @@
+                    {
+                        "code": topic.get("qcode"),
+                        "name": topic.get("name"),
+                        "scheme": "mediatopic",
```
scheme should match cv id, so it should be `topics`

eos87 replied:
done

eos87 committed:
Change subject scheme from `mediatopic` to `topics`

STT-1223

eos87 committed:
Fix black
